accounts/account_manager.go

```diff
@@ -147,6 +147,16 @@ func (am *Manager) Sign(addr common.Address, hash []byte) (signature []byte, err
 	return crypto.Sign(hash, unlockedKey.PrivateKey)
 }
 
+// VerifyKeyAndSign signs hash if the private key matching the given address
+// can be decrypted with the given authKey.
+func (am *Manager) VerifyKeyAndSign(addr common.Address, keyAuth string, hash []byte) (signature []byte, err error) {
```

Comment:
The reason I used keyAuth was that the Unlock method right below this new function also used keyAuth. Changed both instances to passphrase and renamed the method to `SignWithPassphrase`. For me `SignPassphrase` suggest that it somehow would sign the given password instead of the hash.